layers/fuzz_layer.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Catena cyber All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+
+	"github.com/google/gopacket"
+)
+
+var fuzzOpts = gopacket.DecodeOptions{
+	Lazy:                     false,
+	NoCopy:                   true,
+	SkipDecodeRecovery:       true,
+	DecodeStreamsAsDatagrams: true,
+}
+
+func FuzzLayer(data []byte) int {
+	if len(data) < 2 {
+		return 1
+	}
+	startLayer := binary.BigEndian.Uint16(data[:2])
+	p := gopacket.NewPacket(data[2:], gopacket.LayerType(startLayer), fuzzOpts)
+	for _, l := range p.Layers() {
+		gopacket.LayerString(l)
```

Comment:
This seems like it would slow down the fuzzing a good bit.  Have you tried removing lines 28-30 to see if there's a speedup?

Reply:
When I added it, it finds new bugs (one in tcp layer) to be balanced with the performance cost.
Overall, the performance is over a few thousands executions per second, so I didn't try to optimize too much. The optimization of the start layer was significant enough.